crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -52,19 +53,19 @@ pub struct Ethereum {
     web3: DynWeb3,
     network: Network,
     contracts: Contracts,
-    gas: Arc<NativeGasEstimator>,
+    gas: Arc<GasPriceEstimator>,
 }
 
 impl Ethereum {
     /// Access the Ethereum blockchain through an RPC API.
-    pub async fn new(rpc: Rpc, addresses: contracts::Addresses) -> Result<Self, Error> {
+    pub async fn new(
+        rpc: Rpc,
+        addresses: contracts::Addresses,
+        mempools: &[mempool::Config],
```

Comment:
I'm not sure what our dependency injection strategy with DDD is, but I find it weird that `Ethereum` needs to know about our mempool config. This is because the GasPriceEstimator gets created inside the `new` function. This was because before it was only dependent n Web3, which we already pass in as it's widely needed. Given the new dependency, I'd prefer to inject GasPriceEstimator (or additional tip config) directly in the new function rather than coupling mempools and ethereum.

Reply:
Indeed. Moved out.